kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/operator/KroxyliciousOperatorBundleInstaller.java

```diff
@@ -0,0 +1,402 @@
+
+    private static final Logger LOGGER = LogManager.getLogger(KroxyliciousOperatorBundleInstaller.class);
+    private static final String SEPARATOR = String.join("", Collections.nCopies(76, "="));
+
+    private static final KubeClusterResource cluster = KubeClusterResource.getInstance();
+    private static List<File> operatorFiles;
+
+    private final ExtensionContext extensionContext;
+    private String kroxyliciousOperatorName;
+    private String namespaceInstallTo;
+    private String namespaceToWatch;
+    private List<String> bindingsNamespaces;
+    private Duration operationTimeout;
+    private Duration reconciliationInterval;
+    private Map<String, String> extraLabels;
+    private final int replicas;
+
+    private String testClassName;
+    // by default, we expect at least empty method name in order to collect logs correctly
+    private String testMethodName = "";
+
+    private static final Predicate<KroxyliciousOperatorBundleInstaller> IS_EMPTY = ko -> ko.extensionContext == null && ko.kroxyliciousOperatorName == null
+            && ko.namespaceInstallTo == null &&
+            ko.namespaceToWatch == null && ko.bindingsNamespaces == null && ko.operationTimeout == null && ko.reconciliationInterval == null
+            && ko.testClassName == null && ko.testMethodName == null;
+
+    private static final Predicate<File> installFiles = file -> !file.getName().contains("Deployment");
+
+    public KroxyliciousOperatorBundleInstaller() {
+        this.namespaceInstallTo = Constants.KO_NAMESPACE;
+        this.replicas = 1;
+        this.extensionContext = ResourceManager.getTestContext();
+    }
+
+    @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
+    public KroxyliciousOperatorBundleInstaller(KroxyliciousOperatorBuilder builder) {
+        this.extensionContext = builder.extensionContext;
+        this.kroxyliciousOperatorName = builder.kroxyliciousOperatorName;
+        this.namespaceInstallTo = builder.namespaceInstallTo;
+        this.namespaceToWatch = builder.namespaceToWatch;
+        this.bindingsNamespaces = builder.bindingsNamespaces;
+        this.operationTimeout = builder.operationTimeout;
+        this.reconciliationInterval = builder.reconciliationInterval;
+        this.extraLabels = builder.extraLabels;
+        this.replicas = builder.replicas;
+
+        // assign defaults is something is not specified
+        if (this.kroxyliciousOperatorName == null || this.kroxyliciousOperatorName.isEmpty()) {
+            this.kroxyliciousOperatorName = Constants.KO_DEPLOYMENT_NAME;
+        }
+        // if namespace is not set we install operator to 'kroxylicious-operator'
+        if (this.namespaceInstallTo == null || this.namespaceInstallTo.isEmpty()) {
+            this.namespaceInstallTo = Constants.KO_NAMESPACE;
+        }
+        if (this.namespaceToWatch == null) {
+            this.namespaceToWatch = this.namespaceInstallTo;
+        }
+        if (this.bindingsNamespaces == null) {
```

Comment:
In what circumstance are we going to need a other bindings namespaces?

Reply:
removed